### ISSUE TITLE
Change the Serial pins

### ISSUE THREAD

**Issue by fqantonio** [chipKIT32-libraries/Microchip_RN487x#4]
I'm using ATMEGA328p with RN4871 BLE. The serial pins I'm using are the arduino 0,1 ( RX, TX) and not the 25 and 26 of your fubarino. How can I change that? Thank for your work. Best

**Comment by JacobChrist** [comment]
I think the best bet is to connect the BLE to pins 25/26 of the fub, modify
the library to use these pins (if needed).  Can you point to the library
for this module?

Jacob

**Comment by EmbeddedMan** [comment]
I *think* that you can just use the normal Arduino serial object in this line
`#define bleSerial Serial1       // Serial1 is UART2 on pins 26 (TX) and 25 (RX) to BLE module`
In other words, we don't actually set those pins in the library anywhere. You pass into the library a serial object of a certain name (for Fubarino, that's "Serial1"), and that serial object has certain TX and RX pins associated with it (which are not changeable). For every board type these may come out on different pins. 
Is the normal serial port (pins 0 and 1) on an Arduino just "Serial"? If so, then you can just use "Serial" in place of "Serial1" in the examples. That will mean, however, that you will need to disable all of the norma PC-serial stuff, since that's currently using the "Serial" object.

**Comment by EmbeddedMan** [comment]
fquantinio is not using a Fubarino, but rather a normal Arduino. The library should not need any modifications other than just passing in the proper serial object.

**Comment by fqantonio** [comment]
> I _think_ that you can just use the normal Arduino serial object in this line
> `#define bleSerial Serial1 // Serial1 is UART2 on pins 26 (TX) and 25 (RX) to BLE module`
> In other words, we don't actually set those pins in the library anywhere. You pass into the library a serial object of a certain name (for Fubarino, that's "Serial1"), and that serial object has certain TX and RX pins associated with it (which are not changeable). For every board type these may come out on different pins.
> Is the normal serial port (pins 0 and 1) on an Arduino just "Serial"? If so, then you can just use "Serial" in place of "Serial1" in the examples. That will mean, however, that you will need to disable all of the norma PC-serial stuff, since that's currently using the "Serial" object.

Yes, that's the problem. To program the ATMEGA328p I use the FTDI (RX TX pins 0,1). But I also use them, when I don't have the FTDI pluged on to the board, for the BLE RN4871. You say if I use only Serial I will use the common pins 0 and 1?

**Comment by fqantonio** [comment]
> fquantinio is not using a Fubarino, but rather a normal Arduino. The library should not need any modifications other than just passing in the proper serial object.

That's true! But if I also use the same serial to the FTDI (even if it is not pluged into the Board when I'm using the BLE). Use the same serial? FTDI and BLE are exclusives, I use one or the other, not both.

**Comment by EmbeddedMan** [comment]
I think that's right. Just remove all of the 'debug' serial stuff (that's
routed through the "Serial" serial port in the example sketches) and then
use "Serial" for your BLE code.

The trouble is that it's going to be hard for you to debug what's going
wrong with no way to contact the PC with serial prints.

*Brian
